arangod/Aql/Functions.cpp

```diff
@@ -2481,9 +2482,18 @@ AqlValue Functions::Normalize(ExpressionContext* ctx, AstNode const&,
   ::appendAsString(vopts, adapter, value);
   
   dest = TRI_normalize_utf8_to_NFC(buffer->c_str(), buffer->length(), &outlength);
-  auto ret = AqlValue(dest, outlength);
-  delete dest;
-  return ret;
+  if (dest == nullptr) {
+    return AqlValue(AqlValueHintNull());
```

Comment:
This should be a `TRI_ERROR_OUT_OF_MEMORY` exception. There is no point in continuing the query here.